src/main/java/seedu/address/commons/util/StringUtil.java

```diff
@@ -38,6 +38,86 @@ public static boolean containsWordIgnoreCase(String sentence, String word) {
         return preppedSentence.contains(preppedKeyword);
     }
 
+    /**
+     * Returns true if the {@code tagString} contains the {@code word}.
+     *   Ignores case, and a partial or full word match is required.
+     *   <br>examples:<pre>
+     *       tagContainsWordIgnoreCase("ABc def", "abc") == true //a full word match
+     *       tagContainsWordIgnoreCase("ABc def", "DEF") == true //a full word match
+     *       tagContainsWordIgnoreCase("ABc def", "Abc Def") == true //a full word match
+     *       tagContainsWordIgnoreCase("ABc def", "AB") == true //a partial word match
+     *       tagContainsWordIgnoreCase("ABc def", "ABc de") == true //a partial word match
+     *       tagContainsWordIgnoreCase("ABc def", "AB de") == true //a partial word match
+     *       tagContainsWordIgnoreCase("AbC def", "AcB") == false //no match
+     *       </pre>
+     * @param ownerName cannot be null
+     * @param word cannot be null, cannot be empty
+     */
+    public static boolean nameContainsWordsIgnoreCase(String ownerName, String word) {
```

Comment:
Regarding the first point, they are different because `nameContains...` only matches a name (single or multiple words) to a single search word. However, `tagContains...` is a little different as I want it to return true even if user gives multiple words as search words and also if user makes a small error such as typing "Borde Collie" instead of "Border Collie" the method will still return true. This may not work for names as omitting a some letters from a name may result in a completely different person.

Reply:
Regarding the second point, I will use the method to make it neater! Thanks for the suggestion!

Reply:
On second thought I think you are right! It does not matter too much in the different matching of the names or tags after thinking about it and they should be rather similar since we are just finding 'potential matches'! Thanks for pointing it out! Will refactor!